e2etest/GuestProxyAgentTest/LinuxScripts/GuestProxyAgentExtensionValidation.sh

```diff
@@ -0,0 +1,154 @@
+echo "PIRExtensionVersion=$PIRExtensionVersion"
+
+echo "detecting os and installing jq" 
+os=$(hostnamectl | grep "Operating System")
+echo "os=$os"
+if [[ $os == *"Ubuntu"* ]]; then
+    for  i in {1..3}; do
+        echo "start installing jq via apt-get $i"
+        sudo apt update
+        sudo apt-get install -y jq
+        sleep 10
+        install=$(apt list --installed jq)
+        echo "install=$install"
+        if [[ $install == *"jq"* ]]; then
+            echo "jq installed successfully"
+            break
+        fi
+    done
+else
+    for  i in {1..3}; do
+        echo "start installing jq via yum $i"
+        sudo yum -y install jq
+        sleep 10
+        install=$(yum list --installed jq)
+        echo "install=$install"
+        if [[ $install == *"jq"* ]]; then
+            echo "jq installed successfully"
+            break
+        fi
+    done
+fi
+
+echo "Check that status file is regenerated"
+timeout=900
+elpased=0
+while :; do 
+    statusFolder=$(find "$PIRExtensionFolderPath" -type d -name 'status')
+	statusFile=$(ls $statusFolder/*.status)
+	if [ -f "$statusFile" ]; then
+		echo "statusFile=$statusFile"
+        echo "Contents of status file:"
+        cat "$statusFile"
+        statusExists=true
+		break
+    fi
+    ((elapsed += 5))
+    if [[ $elapsed -ge $timeout ]]; then
+		echo "Timeout reached. Exiting the loop, status file is not regenerated."
+        statusExists=false
+		break
+    fi
+	sleep 5
+done
+
+echo "TEST: Check that status file is success with 5 minute timeout"
+guestProxyAgentExtensionStatusObjGenerated=false
+guestProxyAgentExtensionServiceStatus=false
+timeout=300
+elpased=0
+if [[ "$statusExists" == "true" ]]; then
+	while :; do 
+    extensionStatus=$(cat "$statusFile" | jq -r '.[0].status.status')
+    if [[ "$extensionStatus" == "success" ]]; then
+        guestProxyAgentExtensionStatusObjGenerated=true
+        guestProxyAgentExtensionServiceStatus=true
+        echo "The status is success."
+        break
+    fi
+    ((elapsed += 5))
+    if [[ $elapsed -ge $timeout ]]; then
+        echo "Timeout reached. Exiting the loop."
+        break
+    fi
+    sleep 5
+done
+fi
+
+echo "TEST: Check that process ProxyAgentExt is running"
+processId=$(pgrep ProxyAgentExt)
+echo "processId=$processId"
+if [ -z "$processId" ]; then
+    echo "Process ProxyAgentExt is not running"
+    guestProxyAgentExtensionServiceExist=false
+    guestProxyAgentExtensionProcessExist=false
+else 
+    echo "Process ProxyAgentExt is running"
+    guestProxyAgentExtensionServiceExist=true
+    guestProxyAgentExtensionProcessExist=true
+fi
+
+echo Write-Output "TEST: ProxyAgent version running in VM is the same as expected version" 
+proxyAgentVersion="$(eval "$PIRExtensionFolderPath/ProxyAgent/ProxyAgent/azure-proxy-agent --version")"
+echo "proxy agent version from extension folder: $proxyAgentVersion"
+guestProxyAgentExtensionVersionUpgrade=true
+if [ -z "$expectedProxyAgentVersion" ]; then
```

Comment:
Where you use this expectedProxyAgentVersion to compare against extractedVersion or proxyAgentVersion?